kong/reports.lua

```diff
@@ -165,7 +167,12 @@ local function send_report(signal_type, t, host, port)
   -- send return nil plus err msg on failure
   local bytes, err = sock:send(concat(_buffer, ";", 1, mutable_idx) .. "\n")
   if bytes then
-    sock:setkeepalive()
+    local ok, err = sock:setkeepalive()
+    if not ok then
+      log(DEBUG, "failed to keepalive to ", host, ":", tostring(port), ": ", err)
+      _ssl_session = nil  -- force new SSL session on next request
```

Comment:
```suggestion
```

I think `_ssl_session` does not need to clear, it just closes the TCP connection, not need to close the TLS session, Next TCP connection still can reuse this TLS session to accelerate the TLS handshake mechanism.